lib/code-explorer.js

```diff
@@ -0,0 +1,272 @@
+  }
+  // Formatting unprocessed/errored octets without coloring.
+  var groupSpan = null;
+  for (var i = lastPosition; i < octets.length; i++) {
+    var octet = octets[i];
+    if (!groupSpan || groupSpan.parentNode !== octet.parentNode) {
+        groupSpan = document.createElement('span');
+        groupSpan.className = 'grp';
+        octet.parentNode.insertBefore(groupSpan, octet);
+    }
+    groupSpan.appendChild(octet);
+  }
+}
+
+function disassemble() {
+  var text = document.getElementById('text');
+  text.textContent = '';
+  try {
+    var reader = new wasmparser.BinaryReader();
+    reader.setData(content, 0, content.byteLength);
+    var dis = new wasmparser.WasmDisassembler();
+    dis.addOffsets = true;
+    var done = dis.disassembleChunk(reader);
+    var result = dis.getResult();
+    result.lines.forEach(function (s, index) {
+        var line = document.createElement('div');
+        line.className = 'line';
+        var offset = result.offsets[index];
+
+        // ignoring offset for lines with only '(' and ')'
+        var i = 0, j = s.length - 1;
+        while (i < j && s[i] === ' ') i++;
+        while (i < j && s[j] === ' ') j++;
+        if (i === j && (s[i] === '(' || s[i] === ')'))
+            offset = undefined;
+
+        line.textContent = s;
+        if (offset)
+            line.dataset.offset = offset;
+        text.appendChild(line);
+    });
+  } catch (_) {
+    // ignoring error
+  }
+}
+
+function openWasm(rootArr) {
+    var dump = document.getElementById('dump');
+    var addresses = dump.querySelector('.addresses');
+    addresses.textContent = '';
+    var rows = dump.querySelector('.rows');
+    rows.textContent = '';
+    var asciis = dump.querySelector('.asciis');
+    asciis.textContent = '';
+    content = rootArr;
+    var rowCount = Math.max(1, Math.ceil(content.length / perRow));
+    for (var i = 0; i < rowCount; i++) {
+        var row = document.createElement('div');
+        row.className = 'row';
+        var address = document.createElement('div');
+        address.className = 'address';
+        var rowOffset = i * perRow;
+        address.textContent = "0x" + toHex(rowOffset, 8);
+        addresses.appendChild(address);
+        var itemsCount = Math.min(content.length - rowOffset, perRow);
+        var str = '';
+        for (var j = 0; j < itemsCount; j++) {
+            var b = content[rowOffset + j];
+            var octet = document.createElement('span');
+            octet.className = 'o';
+            octet.textContent = toHex(b, 2);
+            row.appendChild(octet);
+            str += b >= 32 && b < 127 ? String.fromCharCode(b) : '.';
+        }
+        rows.appendChild(row);
+        var ascii = document.createElement('div');
+        ascii.className = 'ascii';
+        ascii.textContent = str;
+        asciis.appendChild(ascii);
+    }
+    paintCode(dump.querySelectorAll('.o'));
+    disassemble();
+}
+
+initialize();
+openWasm(window["rootArr"]);
```

Comment:
I have saved Uint8Array in window["rootArr"] (assigned in HTML). Any other better way to access, assign variable?

Reply:
> Any better way to access file data?

In this context, postMessage and message event will work just fine.

Reply:
services.ts:
```
static openBinaryExplorer(file: File) {
    let child_window = null;
    child_window = window.open("./lib/code-explorer.html", "", "toolbar=no,ocation=no,directories=no,status=no,menubar=no,location=no,scrollbars=yes,resizable=yes,width=1024,height=568");
    child_window.postMessage(file.data, "*");
  }
```

code-explorer.js:
```
window.addEventListener('message', function(e) {
	console.log("event happened");
	recievedFileData(e);
} , false);

function recievedFileData(message) {
	console.log(message);
}
```

I am recieving rubbish in message.data:
![Screenshot](https://i.snag.gy/HR2I0b.jpg)

Reply:
it's probably coming from react's dev server. You will need to filter only one you care, e.g. add the type property on the object you are sending.

Reply:
Euhm...what about data is not being received. No message containing file binary is visible in my logs.

Reply:
use load events to track when other window is load, or notify the parent with postMessage when it's ready for data